docs/06/01/setup-guide.md

```diff
@@ -0,0 +1,255 @@
+  DatePicker: props => <MyCustomDatePicker {...props} />,
+  Heading: props => <MyCustomHeading {...props} />,
+  Link: props => <MyCustomLink {...props} />,
+  Menu: props => <MyCustomMenu {...props} />,
+  NumberInput: props => <MyCustomNumberInput {...props} />,
+  OrderedList: props => <MyCustomOrderedList {...props} />,
+  ProgressBar: props => <MyCustomProgressBar {...props} />,
+  Radio: props => <MyCustomRadio {...props} />,
+  RadioGroup: props => <MyCustomRadioGroup {...props} />,
+  Select: props => <MyCustomSelect {...props} />,
+  Switch: props => <MyCustomSwitch {...props} />,
+  Table: props => <MyCustomTable {...props} />,
+  Text: props => <MyCustomText {...props} />,
+  TextInput: props => <MyCustomTextInput {...props} />,
+  UnorderedList: props => <MyCustomUnorderedList {...props} />,
+}
+```
+
+### 3. Use GustoProviderCustomUIAdapter
+
+Instead of using the standard `GustoApiProvider`, use the `GustoProviderCustomUIAdapter` ([View on GitHub](https://github.com/Gusto/embedded-react-sdk/blob/main/src/contexts/GustoProvider/GustoProviderCustomUIAdapter.tsx)) to provide your custom components:
+
+```tsx
+import { GustoProviderCustomUIAdapter } from '@gusto/embedded-react-sdk'
+
+function App() {
+  return (
+    <GustoProviderCustomUIAdapter
+      config={{ baseUrl: '/api/gusto/' }}
+      components={myCustomComponents}
+    >
+      {/* Your application components */}
+      <EmployeeOnboardingFlow companyId="company_123" />
+    </GustoProviderCustomUIAdapter>
+  )
+}
+```
+
+### 4. Implement Only the Components You Need
+
+You don't need to implement every component in the `ComponentsContextType` interface. You can choose to implement only the ones you want to customize and let the default components handle the rest by importing the `defaultComponents` ([View on GitHub](https://github.com/Gusto/embedded-react-sdk/blob/main/src/contexts/ComponentAdapter/adapters/defaultComponentAdapter.tsx)):
+
+```tsx
+import { GustoProviderCustomUIAdapter, defaultComponents } from '@gusto/embedded-react-sdk'
+
+// Merge your custom components with the defaults
+const customComponents = {
+  ...defaultComponents,
+  Button: props => <MyCustomButton {...props} />,
+  TextInput: props => <MyCustomTextInput {...props} />,
+}
+
+function App() {
+  return (
+    <GustoProviderCustomUIAdapter config={{ baseUrl: '/api/gusto/' }} components={customComponents}>
+      {/* Your application components */}
+    </GustoProviderCustomUIAdapter>
+  )
+}
+```
+
+> **Important Note**: When using the partial implementation approach (spreading `defaultComponents`), be aware that the default components include React Aria design system dependencies. This can potentially increase your bundle size and prevent React Aria from being properly tree-shaken away, which goes against our optimization goals. For optimal bundle size, consider implementing all required components with your own design system instead of relying on the default implementations.
```

Comment:
Same here, this is the disclaimer that should potentially be made for `GustoProvider` usage and why they might consider using `GustoProviderCustomUIAdapter` instead

Reply:
Kk, I removed this, I think i've covered it enough above